plugin/src/main/scala/sbtcompendium/CompendiumPlugin.scala

```diff
@@ -53,19 +58,23 @@ object CompendiumPlugin extends AutoPlugin {
     compendiumProtocolIdentifiers := Nil,
     compendiumServerHost := "localhost",
     compendiumServerPort := 47047,
+    compendiumFormatSchema := "avro",
     compendiumGenClients := {
       val log = sbt.Keys.streams.value.log
 
-      val generateProtocols = compendiumProtocolIdentifiers.value.toList.map { protocolId =>
-        val targetFile = (sbt.Keys.sourceManaged in Compile).value / "compendium" / s"$protocolId.scala"
-        val generateProtocol = CompendiumUtils.generateCodeFor(
-          protocolId,
-          targetFile,
-          client(compendiumServerHost.value, compendiumServerPort.value).generateClient
-        )
-
-        log.info(s"Attempting to generate client for [${protocolId}]")
-        generateProtocol
+      val generateProtocols = compendiumProtocolIdentifiers.value.toList.map {
+        protocolId =>
+          def targetFile(id: String) =
+            (sbt.Keys.sourceManaged in Compile).value / "compendium" / s"$protocolId$id.scala"
+          val generateProtocol = CompendiumUtils.generateCodeFor(
+            protocolId,
+            targetFile,
+            client(compendiumServerHost.value, compendiumServerPort.value).generateClient,
```

Comment:
Maybe we can extract the client creation outside of the `toList.map`, so we'd only need to create one client per compilation cycle (that's my bad :D )

Reply:
sure